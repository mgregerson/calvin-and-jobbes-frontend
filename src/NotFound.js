import "./NotFound.css";

/** NotFound
 *
 * Renders 404 page
 *
 * RoutesList -> NotFound
 */

function NotFound() {
  return (
    <img
      className="NotFound"
      src="https://ubiq.co/tech-blog/wp-content/uploads/2020/11/How-to-Make-a-Custom-404-Page-730x410.jpg"
    ></img>
  );
}

export default NotFound;
