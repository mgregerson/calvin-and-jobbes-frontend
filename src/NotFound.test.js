import NotFound from "./NotFound";
import { render } from "@testing-library/react";

describe("NotFound Component", function () {
  it("renders without crashing", function () {
    render(<NotFound />);
  });
  test("matches snapshot", function () {
    const { container } = render(<NotFound />);
    expect(container).toMatchSnapshot();
  });
  it("It displays the 404 image on the page", function () {
    const { container } = render(<NotFound />);
    expect(container.querySelector(".NotFound-photo")).toBeInTheDocument();
  });
  it("Displays a Not Found message", function () {
    const { container } = render(<NotFound />);
    expect(container.querySelector(".NotFound-message").innerHTML).toEqual(
      "Sorry, we couldn't find that page!"
    );
  });
});
