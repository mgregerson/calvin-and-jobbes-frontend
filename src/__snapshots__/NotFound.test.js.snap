// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component matches snapshot 1`] = `
<div>
  <div
    class="NotFound"
  >
    <img
      class="NotFound-photo"
      src="../404img.jpg"
    />
    <h3
      class="NotFound-message"
    >
      Sorry, we couldn't find that page!
    </h3>
  </div>
</div>
`;
